dolphin/db/sqlalchemy/api.py

```diff
@@ -143,12 +245,22 @@ def storage_update(context, storage_id, values):
 
 def storage_get(context, storage_id):
     """Retrieve a storage device."""
-    this_session = get_session()
-    this_session.begin()
-    storage_by_id = this_session.query(Storage) \
-        .filter(Storage.id == storage_id) \
-        .first()
-    return storage_by_id
+    return _storage_get_get(context, storage_id)
+
+
+def _storage_get_get(context, storage_id, session=None):
```

Comment:
What is this naming, storage_get_get? why can't it be simply get_storage or storage_get